### DISCUSSION TITLE
Feature Request: Enhance Open with Other PDF Reader

### DISCUSSION THREAD

**Discussion by m2kar** [sumatrapdfreader/sumatrapdf#2474] [Ideas]
Current "Open With" feature only support few third-party reader like AdobePDF, and don't support other pdf reader like drawboard pdf, etc. 

![image](https://user-images.githubusercontent.com/16930652/162704239-0d80958d-a9ad-4621-bd7f-0d2506775d9f.png)

In windows, the "Open With" menu include other PDF reader, which could be included in SumatraPDF's "open with" List

![image](https://user-images.githubusercontent.com/16930652/162708479-db51ab3f-592e-4f32-821d-47d039e1b1d5.png)

**Comment by GitHubRulesOK** [comment]
You can add your own its in the docs, but when I tried adding all 20 or so that I use for tests etc. I found it easier to add just one for native Edge and send anything else to a batch file that throws up the full set on a taskbar with icons. the standard ones dont show portable apps so I include Acrobat/Foxit/Tracker portable on that toolbar too

**Comment by m2kar** [comment]
Thanks, I found the setting docs in the official mannual. 

https://www.sumatrapdfreader.org/docs/Customize-external-viewers

**Reply by GitHubRulesOK** [reply]
What is not highlighted in that doc is `Name = Foxit &Basic 3D view or Review Text` means that for english (may differ by menu language) the **&** will allow shortcut as ALT+F+**B** in this case but beware many others are already assigned so a limited set from a-z and symbol keys are free for your unique assignment.

**Comment by m2kar** [comment]
I support add the Windows's OpenAs Dialog to the default config.

I have done it in self-defined config.

```
# SumatraPDF-settings.txt
ExternalViewers [
、	# windows OpenAs Dialog

	[
		CommandLine = C:\Windows\system32\cmd.exe /c start /min "" powershell -WindowStyle Hidden -ExecutionPolicy Bypass /c rundll32.exe Shell32.dll,OpenAs_RunDLL "%1"
		Name = Windows OpenAs Dialog
		Filter = *.pdf
	
	]
		
]
```

And the effect is: 

![image](https://user-images.githubusercontent.com/16930652/163205645-5c04993d-43bc-480a-b005-b009a7fc5516.png)

![image](https://user-images.githubusercontent.com/16930652/163205740-4732044e-627d-478f-b9ba-4c13fb58e826.png)

May be a good idea :)

**Reply by GitHubRulesOK** [reply]
Good idea perhaps, but not in my case
 I am in restricted settings where that command will work in console but not allowed from SumatraPDF and I tried different methods.

 ALSO the apps are not in the system unless previously run, they can be on different USB drives each time the system is  restarted day to day or even twice in the day. Hence I rely on toolbars that update on restarts.

 I sometimes use a command line sidebar file launcher to call other apps with the current files.
This is usually vertical parked on the left but for this comment its horizontal so I can send one current tabbed file to many apps for testing.
![image](https://user-images.githubusercontent.com/12754913/163290771-e34b9f2b-6e69-4ca6-a069-f57f7bacfcec.png)

**Reply by m2kar** [reply]
I know what you mean. Here is my workflow. SumatraPDF is fast than another pdf reader, so I set it as the default PDF reader. In this way, I can open pdf files from Wechat, Zotero, and Telegram faster . But I often need to make complicated annotations and other operations with another PDF reader, like Foxit PhantomPDF and Drawboard PDF. But 
unfortunately, Drawboard PDF is a UWP application and **do not support open files with commands**. So I need a compromised method to open the file with Drawboard PDF. 
In my old way, I use "File" -> "Open in Folder" -> "Find the file" -> "Right Click" -> "Click OpenAS menu"  -> "select Drawboard PDF", which is a long work flow.
In my new work flow, I just need to "File" -> "Windows OpenAs &Dialog" -> "select Drawboard PDF", and can be simplified to "ALT+F+D" ->"select Drawboard PDF".

**Reply by GitHubRulesOK** [reply]
UWP !! That is a good reason for the long command Personally I don't think I have any loaded on this machine (rarely on my others) MS Store is the first bloat-ware deleted along with some others like one drive or games (except solitaire in case I get bored)

**Reply by kjk** [reply]
@m2kar  I would like to improve annotations editing in SumatraPDF and since you're annotation user it would be great if you could comment on https://github.com/sumatrapdfreader/sumatrapdf/issues/2485 with list of the most important annotation-related features that SumatraPDF is missing.

**Reply by m2kar** [reply]
> UWP !! That is a good reason for the long command Personally I don't think I have any loaded on this machine (rarely on my others) MS Store is the first bloat-ware deleted along with some others like one drive or games (except solitaire in case I get bored)

Yes. I have tried many pdf readers, but the Drawboard PDF is the best for touch screens and touch pens on the windows platform, even though it's slower than Sumatra.

**Reply by m2kar** [reply]
> @m2kar I would like to improve annotations editing in SumatraPDF and since you're annotation user it would be great if you could comment on #2485 with list of the most important annotation-related features that SumatraPDF is missing.

OK, no problem. 😊
